packages/grid/test/frozen-columns.test.js

```diff
@@ -291,6 +291,24 @@ const transformsEqual = (element, transform) => {
           expect(cells[1].hasAttribute('first-frozen-to-end')).to.be.true;
           expect(cells[2].hasAttribute('first-frozen-to-end')).not.to.be.true;
         });
+
+        it('should stay frozen to end when resize results in total width less than the grid', async () => {
+          grid._columnTree[0][2].flexGrow = 0;
+
+          grid.style.width = '400px';
+          await onceResized(grid);
+
+          const initBoundingClientRect = getRowCells(containerRows[0])[2].getBoundingClientRect();
+
+          grid._columnTree[0][0].width = '10px';
+          grid._columnTree[0][0].flexGrow = 0;
+          grid._columnTree[0][1].width = '10px';
+          grid._columnTree[0][1].flexGrow = 0;
+
+          const finalBoundingClientRect = getRowCells(containerRows[0])[2].getBoundingClientRect();
+          expect(finalBoundingClientRect.x).to.equal(initBoundingClientRect.x);
+          expect(finalBoundingClientRect.width).to.equal(initBoundingClientRect.width);
+        });
       });
     });
```

Comment:
Currently, if you click the space between the columns, you get JS errors:

https://user-images.githubusercontent.com/1222264/201299379-4fb17367-339f-4ec3-ba11-5bdc9db02a80.mp4

Technically, this was an issue even before this PR but it becomes more apparent with the change (for example when clicking a grid row selects an item).

`getEventContext(clickEvent)` for a `<tr>` click should produce a context with `item`, `section: 'body'`, `index`, `column: undefined`...